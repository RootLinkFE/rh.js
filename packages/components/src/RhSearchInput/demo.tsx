import RhSearchInput from '.';
import React from 'react';
import { message } from 'antd';

function Demo() {
  return (
    <div style={{ width: 300 }}>
      <RhSearchInput
        size="small"
        suffix={null}
        onSearch={(v) => {
          console.log(v);
          message.info('查询' + v);
        }}
      />
      <br />
      <br />
      <br />
      <RhSearchInput
        size="large"
        onSearch={(v) => {
          console.log(v);
          message.info('查询' + v);
        }}
      />
      <br />
      <br />
      <br />
      <RhSearchInput
        bordered={false}
        placeholder="无边框展示"
        onSearch={(v) => {
          console.log(v);
          message.info('查询' + v);
        }}
      />
    </div>
  );
}

export default Demo;
