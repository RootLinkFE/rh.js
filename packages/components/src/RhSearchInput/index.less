@import '../styles/index.less';

.rh-search-input {
  &.ant-input-affix-wrapper-borderless {
    padding: 4px 2px;
    border-bottom: 1px solid #d4d9e2 !important;
    box-shadow: none !important;
  }
  &.ant-input-affix-wrapper-focused {
    &.ant-input-affix-wrapper-borderless {
      border-bottom: 2px solid @primary-color !important;
    }
  }
}
