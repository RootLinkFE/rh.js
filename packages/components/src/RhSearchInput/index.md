---
title: RhSearchInput
nav:
  title: 'Components'
  path: /components
group:
  title: '数据展示'
---

# RhSearchInput

搜索输入框

- 自定义 Icon
- 防抖触发回车事件、搜索图标点击事件
- border 样式控制

## Demo

<code src="./demo.tsx"/>

<API/>
