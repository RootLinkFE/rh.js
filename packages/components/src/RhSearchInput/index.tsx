import { SearchOutlined } from '@ant-design/icons';
import { useDebounceFn } from 'ahooks';
import { Input } from 'antd';
import React, { useState } from 'react';
import './index.less';

export type RhSearchInputProps = {
  /**
   * 是否显示border
   */
  bordered?: boolean;
  /**
   * 输入提示
   */
  placeholder?: string;
  /**
   * 防抖延迟时间（ms)
   * @default 200
   */
  delayTime?: number;

  /**
   * 搜索回调，仅回车和图标点击时触发
   */
  onSearch?: (v: string) => void;
  /**
   * change回调，文字变化即触发
   */
  onChange?: (v: string) => void;
  [key: string]: any;
};

function RhSearchInput({
  placeholder = '请输入',
  bordered = true,
  delayTime = 200,
  onSearch = () => {},
  onChange = () => {},
  ...restProps
}: RhSearchInputProps) {
  const [value, setValue] = useState<string>('');

  const { run: onSearchRun } = useDebounceFn(
    async () => {
      onSearch(value);
    },
    { wait: delayTime },
  );

  const { run: onChangeRun } = useDebounceFn(
    async () => {
      onChange(value);
    },
    { wait: delayTime },
  );

  return (
    <Input
      allowClear
      value={value}
      placeholder={placeholder}
      bordered={bordered}
      className="rh-search-input"
      onChange={(e) => {
        const v = e.target?.value;
        setValue(v);
        onChangeRun();
      }}
      onKeyPress={(e: React.KeyboardEvent<HTMLInputElement>) => {
        if (e.key === 'Enter') {
          onSearchRun();
        }
      }}
      {...restProps}
      suffix={
        restProps?.suffix || restProps?.suffix === null ? (
          restProps?.suffix
        ) : (
          <SearchOutlined
            onClick={() => {
              onSearchRun();
            }}
            style={{ cursor: 'pointer', fontSize: 22, color: '#9EA5B2' }}
          />
        )
      }
    />
  );
}

export default RhSearchInput;
