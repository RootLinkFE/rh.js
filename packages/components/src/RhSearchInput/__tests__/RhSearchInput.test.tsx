import React from 'react';
import TestRenderer from 'react-test-renderer';
import { createRenderer } from 'react-test-renderer/shallow';
import RhSearchInput from '../index';

it('RhSearchInput renders correctly', () => {
  const renderer = createRenderer();

  renderer.render(
    <RhSearchInput
      size="small"
      suffix={null}
      onSearch={(v) => {
        console.log(v);
      }}
    />,
  );
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
