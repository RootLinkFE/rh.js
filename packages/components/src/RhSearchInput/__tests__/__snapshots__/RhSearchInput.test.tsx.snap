// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RhSearchInput renders correctly 1`] = `
<Input
  allowClear={true}
  bordered={true}
  className="rh-search-input"
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="请输入"
  size="small"
  suffix={null}
  type="text"
  value=""
/>
`;
